app/models/container_project.rb

```diff
@@ -69,4 +70,30 @@ def disconnect_inv
     self.deleted_on = Time.now.utc
     save
   end
+
+  def add_role_to_user(user_name, role_name)
+    if role_name.empty?
+      raise MiqException::MiqProvisionError, "When adding a role to a user the role cannot be empty."
+    end
+    client = ext_management_system.connect
+    client.discover unless client.discovered
+
+    role_binding = Kubeclient::Resource.new
+    role_binding.metadata = {}
+    role_binding.metadata.namespace = name
+    role_binding.metadata.name = role_name
+    role_binding.roleRef = {}
+    role_binding.roleRef.name = role_name
+    role_binding.userNames = [user_name]
+
+    begin
+      client.create_role_binding(role_binding.to_h)
```

Comment:
can you call container_resource_parent_mixin.create_resource here ?

That would allow to remove client related code from here